diff --git a/libavformat/utils.c b/libavformat/utils.c
index 7a74e63..787cb9f 100644
--- a/libavformat/utils.c
+++ b/libavformat/utils.c
@@ -44,6 +44,7 @@
 #include "network.h"
 #endif
 #include "os_support.h"
+#include "fio.h"
 #include "urldecode.h"
 
 /**
@@ -442,13 +443,13 @@ int ff_mkdir_p(const char *path)
         if (*pos == '/' || *pos == '\\') {
             tmp_ch = *pos;
             *pos = '\0';
-            ret = mkdir(temp, 0755);
+            ret = fio_mkdir(temp, 0755);
             *pos = tmp_ch;
         }
     }
 
     if ((*(pos - 1) != '/') && (*(pos - 1) != '\\')) {
-        ret = mkdir(temp, 0755);
+        ret = fio_mkdir(temp, 0755);
     }
 
     av_free(temp);
